submissions/AsyaYeromina/Tiny-JS-world-OOP/index.js

```diff
@@ -0,0 +1,146 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+  Complete the below for code reviewers' convenience:
+
+  Code repository: https://github.com/AsyaYeromina/a-tiny-JS-world
+  Web app: https://asyayeromina.github.io/a-tiny-JS-world/.
+  */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Inhabitant {
+  constructor(inhabitantProperties) {
+    this.species = inhabitantProperties.species;
+    this.name = inhabitantProperties.name;
+    this.gender = inhabitantProperties.gender;
+    this.legs = inhabitantProperties.legs;
+    this.hands = inhabitantProperties.hands;
+    this.speech = inhabitantProperties.speech;
+    this.friends = inhabitantProperties.friends;
+  }
+
+  toString() {
+    return [
+      this.species,
+      this.name,
+      this.gender,
+      this.legs,
+      this.hands,
+      this.speech,
+      this.friends,
+    ]
+      .filter((propertyValue) => propertyValue !== null)
```

Comment:
You won't need `filter` once this class gets rid of props that aren't inherited by all descendants.

Reply:
Yes, thanks
Removed filter